---
output:
  pdf_document:
    toc: no
  word_document:
    fig_caption: yes
    fig_height: 3
    fig_width: 2
    toc: yes
---

# Predictive modeling using epigenomic annotations

Spiro Stilianoudakis^1^ , Mikhail G. Dozmorov^1^ 

^1^ Dept. of Biostatistics, Virginia Commonwealth University, Richmond, VA, 23298, USA 

**Background.** Studies using chromosome conformation capture techniques have shown that the three-dimensional (3D) structure of the genome is folded into distinct compartments, known as topologically associating domains (TADs). TADs serve as units of coordinated gene expression where regulatory elements tend to interact more frequently. The location of TAD boundaries is highly conserved, suggesting the presence of epigenomic marks aiding in TAD formation. The ability to predict which epigenomic features are most associated with the TAD boundaries will allow us to further understand the regulatory role of the 3D structure of the genome. 

**Methods.** Few methods have been developed to model and predict the association between epigenomic features and the TAD boundaries. Moreover, of the existing methods, the models used ignore key characteristics of the data that may hinder the performance of certain parametric models. Specifically, the number of TAD boundaries is much less than the number of other genomic regions, leading to heavily imbalanced classes. Furthermore, most methods utilize direct overlap as a means to quantify the association, while distance, the measure of spatial relationships, remains unaccounted for. Consequently, distances on genomic scale vary widely, leaving uncertainty how the heavily right-tailed distribution of distance measures will affect the model's performance. We propose a novel data pre-processing pipeline that addresses those questions. It includes an elastic-net regularized classification model, where the mixing proportion, \alpha, and the penalization parameter, \lambda, are tuned for over a grid of values [!!! Continue]

**Results.** It was found that the elastic-net model, when applied to pre-processed data, out performed multiple logistic regression models, with and without LASSO-regularized coefficients. Likewise, the variable importance of the elastic-net yielded more stable results compared to the coefficients of the multiple logistic regression models.

**Conclusions.** Current methods used to identify the key molecular drivers of TAD boundaries are not robust enough to handle certain characteristics of genomic data. Models applied to un-processed data of this sort can have poor predictive performances that are masked by the use of performance metrics that do not consider precision or recall statistics, such as ROC curves and subsequent AUCs. The elastic-net model, with its dual use of the \mathcal{l}_{1} and \mathcal{l}_{2} norms allows for both the shrinkage of coefficients to 0, while also being able to handle multiple correlated features. This results in better predictive performances and more accurate identifiction of important features associated with the formation of TAD boundaies.
